### ISSUE TITLE
Better ubuntu (apt-get) install instructions

### ISSUE THREAD

**Issue by ataiya** [glfw/glfw#808]
I know how to compile from sources, but it would be great if you could provide instructions on how to install glfw3 (system/root level) by leveraging the debian packages (ubuntu).

It seems there are some repositories containing glfw3, but from this page it is not clear which repositories should be added: https://launchpad.net/ubuntu/+source/glfw3

This stackoverflow post also discusses similar issues, but there is not straightforward solution that was identified: http://stackoverflow.com/questions/17768008/how-to-build-install-glfw-3-and-use-it-in-a-linux-project

This is another github issue that seems to address a similar problem:
https://github.com/filipwasil/fillwave/issues/58

As you can see, it seems to be a pretty widespread question! :)

**Comment by ataiya** [comment]
By the way, this "works" but I don't love the fact I rely on an independent "keithw" repository!

(Tested on a virgin copy of Ubuntu 16.04)
Add the following line to /etc/apt/sources.list
```
deb http://ppa.launchpad.net/keithw/glfw3/ubuntu trusty main
```

Then execute these commands
```
sudo apt-get update
sudo apt-get install libglfw3
sudo apt-get install libglfw3-dev
```

The library should now be available here:
```
/usr/lib/x86_64-linux-gnu/libglfw.so
/usr/lib/x86_64-linux-gnu/libglfw.so.3
/usr/lib/x86_64-linux-gnu/libglfw.so.3.1
```

And the headers here:
```
/usr/include/GLFW/glfw3.h
```

**Comment by jcowgill** [comment]
The first link you posted is to the main Ubuntu repositories. If you have 14.10 or later and you only want a shared library, you  can just run this to get glfw3:

```
sudo apt-get install libglfw3-dev
```

You don't need any extra repositories.

**Comment by ataiya** [comment]
Right, I guess I assumed I needed this one, which was not on the repo. Nonetheless, a google search didn't pop up anything on how to get glfw3 installed on Ubuntu beside what I show above. Perhaps you should include what you say above on your official page?
 
```sudo apt-get install libglfw3```

**Comment by elmindreda** [comment]
The main reason there aren't any package installation instructions any more is that many distros have often lagged far behind the current stable version and some (not mentioning any names of Fedora) used to ship with random, buggy pre-release snapshots.

Using the distro package is fine if one is making FOSS that will also be packaged or is making personal projects where stability and transferability aren't important. Instead I try to make GLFW easy to build as part of a larger project, like fillwave is doing.

**Comment by ataiya** [comment]
@elmindreda I am not saying that having a great build-from-sources system is not good, but having clear instructions on how to fetch it in standard distros simplifies people's life substantially.
When you have 20+ libraries to install, compiling each from sources is cumbersome. 

Further, I use this library for student assignment/projects, and as they are not very experienced C++ developers, a non packaged library can create a substantial problem...

**Comment by elmindreda** [comment]
I will look into whether the current set of packages are worth mentioning for 3.3.

**Comment by kevinkit** [comment]
+1

**Comment by skippa** [comment]
I'm using Elementary Loki, and I need to install glfw 3.2.1 or greater, however I can't as it says the newest one is installed.  How do I force the installation of the latest one, not the latest one as far as the Elementary OS repo's say?

**Comment by zheyuanWang** [comment]
if I download the zip from https://www.glfw.org/

how can I let it be properly imported then? I found the code in the binding tool of glfw the library is supposed to be imported with 
```
from .library import glfw as _glfw
```

Does it mean that if I unzip the downloaded zip file in the right place, the glfw could find its library?

**Comment by qyalexwen** [comment]
After installation, I found these files:
$ ls libglfw*
libglfw.so  libglfw.so.3  libglfw.so.3.2

But when I tried to link files, it still can not find glfw3: 

$ g++ main.cpp -lglfw3
/usr/bin/ld: cannot find -lglfw3
collect2: error: ld returned 1 exit status

The main.cpp is from this link: 
https://www.glfw.org/documentation.html

**Comment by ataiya** [comment]
```
sudo updatedb #< wait a few minutes
locate libglfw #< find out where
export LD_LIBRARY_PATH="" #< folder where locate found it
```
Cheers

**Comment by tombsar** [comment]
@qyalexwen For linking to the shared library (*.so) you should use `-lglfw`, not `-lglfw3`.

**Comment by twinblasterrex** [comment]
how do I do this for fedora linux

**Comment by lucasmsoares96** [comment]
I needed to use both flags to work on the site example `-lglfw -lGL`
<https://www.glfw.org/docs/latest/quick.html>
